### PR TITLE
feat(typeorm): query builder sql parameters

### PR DESCRIPTION
**Pull request by mottibec**
The attr `db.statement.parameters` has been added to the typeorm query builder instrumentation

### CODE REVIEW

File: packages/instrumentation-typeorm/src/typeorm.ts [hunk 1]
```diff
@@ -201,6 +202,7 @@ export class TypeormInstrumentation extends InstrumentationBase<typeof typeorm>
                     [SemanticAttributes.DB_OPERATION]: operation,
                     [SemanticAttributes.DB_STATEMENT]: sql,
                     [SemanticAttributes.DB_SQL_TABLE]: mainTableName,
+                    [ExtendedDatabaseAttribute.DB_STATEMENT_PARAMETERS]: JSON.stringify(parameters),
```
Comment (blumamir) [inline packages/instrumentation-typeorm/src/typeorm.ts:205]: As this is not in the spec, might contain sensitive information, and can potentially contain large size payloads, I think it's better to add it in a hook, so user can choose a strategy for how to handle it.
WDYT?

Comment (mottibec) [reply packages/instrumentation-typeorm/src/typeorm.ts:205]: I think the user expects to see his SQL query parameters in the instrumentation by default. and that's also what our customers expect and the motivation for this PR

File: packages/instrumentation-typeorm/src/types.ts [hunk 2]
```diff
@@ -23,4 +27,6 @@ export interface TypeormInstrumentationConfig extends InstrumentationConfig {
      * To instrument those set this to `true`
      */
     enableInternalInstrumentation?: boolean;
+    /** set to `true` if you want to instrument the SQL parameters (may leak sensitive information) */
+    collectParameters?: boolean;
```
Comment (blumamir) [inline packages/instrumentation-typeorm/src/types.ts:31]: add to README

File: packages/instrumentation-typeorm/src/types.ts [hunk 3]
```diff
@@ -23,4 +27,6 @@ export interface TypeormInstrumentationConfig extends InstrumentationConfig {
      * To instrument those set this to `true`
      */
     enableInternalInstrumentation?: boolean;
+    /** set to `true` if you want to instrument the SQL parameters (may leak sensitive information) */
```
Comment (blumamir) [inline packages/instrumentation-typeorm/src/types.ts:30]: "if you want to capture the parameter values for parameterized queries (may leak...)"

File: packages/instrumentation-typeorm/src/typeorm.ts [hunk 4]
```diff
@@ -202,6 +203,9 @@ export class TypeormInstrumentation extends InstrumentationBase<typeof typeorm>
                     [SemanticAttributes.DB_STATEMENT]: sql,
                     [SemanticAttributes.DB_SQL_TABLE]: mainTableName,
                 };
+                if (self._config.collectParameters) {
+                    attributes[ExtendedDatabaseAttribute.DB_STATEMENT_PARAMETERS] = JSON.stringify(parameters);
```
Comment (blumamir) [inline packages/instrumentation-typeorm/src/typeorm.ts:207]: can this fail?

**Review by blumamir — APPROVED** [review]

**Pull request merged**